reframe/utility/sanity.py

```diff
@@ -401,13 +430,30 @@ def assert_not_found(patt, filename, msg=None, encoding='utf-8'):
     :returns: ``True`` on success.
     :raises reframe.core.exceptions.SanityError: if assertion fails.
     '''
+    with _open(filename, 'rt', encoding=encoding) as fp:
+        return assert_not_found_s(
+            patt, fp.read(), msg or f'pattern {patt!r} found in {filename!r}'
+        )
+
+
+@deferrable
+def assert_not_found_s(patt, string, msg=None):
+    '''Assert that regex pattern ``patt`` is not found in ``string``.
+
+    This is the inverse of :func:`assert_found_s()`.
+
+    :returns: ``True`` on success.
+    :raises reframe.core.exceptions.SanityError: if assertion fails.
```

Comment:
Here as well, I am not sure how strict we are with documenting all the arguments